api/elastic.ts

```diff
@@ -82,13 +101,24 @@ export const getTransactionBySnapshot = (es: Client) => (term: string): TaskEith
     return findAll(getByFieldQuery<Transaction>(ESIndex.Transactions, 'snapshotHash', term, -1)(es))
 }
 
-const findOne = (search: TransportRequestPromise<ApiResponse>) =>
-    pipe(
-        findAll(search),
-        map(s => s[0])
-    )
+export const getTransactionByAddress = (es: Client) => (term: string, field: 'receiver' | 'sender' | null = null): TaskEither<ApplicationError, Transaction[]> => {
```

Comment:
If you want to keep it type safe, you can combine `keyof` and `Pick` there:
```
field: (keyof Pick<Transaction, 'sender' | 'receiver'>) | null
```

---

`Pick<Transaction, 'sender' | 'receiver'>` gives you a type:
```
{ sender: .., receiver: .. }
```
and using `keyof` on this gives you `'sender' | 'receiver'`